### PR TITLE
[BCF-3381] - Expose latest heads

### PR DESCRIPTION
**Pull request by ilija42**
core ref: 9eb801069a307d5495036f8f716ff277db95b4be
- [common PR](https://github.com/smartcontractkit/chainlink-common/pull/760)
- [core PR](https://github.com/smartcontractkit/chainlink/pull/14394)
- [cosmos PR](https://github.com/smartcontractkit/chainlink-cosmos/pull/419)
- [starknet PR](https://github.com/smartcontractkit/chainlink-starknet/pull/521)
- [chainlink-ccip](https://github.com/smartcontractkit/chainlink-ccip/pull/116)

### CODE REVIEW

**Review by Farber98 — DISMISSED** [review]

File: pkg/solana/chain.go [hunk 1]
```diff
@@ -232,7 +233,38 @@ func newChain(id string, cfg *config.TOMLConfig, ks loop.Keystore, lggr logger.L
 	return &ch, nil
 }
 
-// ChainService interface
+func (c *chain) LatestHead(_ context.Context) (relaytypes.Head, error) {
+	sc, err := c.getClient()
+	if err != nil {
+		return types.Head{}, err
```
Comment (Farber98) [inline pkg/solana/chain.go:238]: We have two times the same import: 
```
"github.com/smartcontractkit/chainlink-common/pkg/types"
relaytypes "github.com/smartcontractkit/chainlink-common/pkg/types"
```

That creates confusion when returning `types.Head{}` in some cases and `relaytypes.Head{}` in others

**Review by kidambisrinivas — DISMISSED** [review]

File: pkg/solana/chain.go (hunk 1 above)
Comment (ilija42) [reply pkg/solana/chain.go:238]: weird how linter allowed this

**Review by aalu1418 — DISMISSED** [review]
lgtm! but seems like there needs to be bumps in other repos for tests to pass

File: pkg/solana/chain.go [hunk 2]
```diff
@@ -232,29 +232,60 @@ func newChain(id string, cfg *config.TOMLConfig, ks loop.Keystore, lggr logger.L
...
 
-// ChainService interface
-func (c *chain) GetChainStatus(ctx context.Context) (relaytypes.ChainStatus, error) {
+func (c *chain) LatestHead(_ context.Context) (types.Head, error) {
+	sc, err := c.getClient()
+	if err != nil {
+		return types.Head{}, err
+	}
+
+	latestBlock, err := sc.GetLatestBlock()
```
Comment (EasterTheBunny) [inline pkg/solana/chain.go:241]: There is a way to specify the commitment level if we need. I'm not sure which level this defaults to and maybe we don't need to specify.

Comment (aalu1418) [reply pkg/solana/chain.go:241]: it just uses the default commitment in the config: https://github.com/smartcontractkit/chainlink-solana/blob/5b8d287fd054c42d22c1362fbcdef9f3622f505d/pkg/solana/client/client.go#L257

which should be "confirmed"

**Review by Farber98 — DISMISSED** [review]

**Review by EasterTheBunny — APPROVED** [review]

**Review by Farber98 — APPROVED** [review]
lgtm

**Review by aalu1418 — APPROVED** [review]

**Pull request merged**